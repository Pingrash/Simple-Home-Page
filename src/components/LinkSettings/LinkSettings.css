.settings-container {
  width: 70%;
  display: flex;
  flex-direction: column;
  flex: 1;
  padding: 0 3rem 1rem;
  background-color: #e6e6e6;
  align-self: center;
  box-shadow: 3px 2px #e1e1e1;
  margin-bottom: 1rem;
}

.cards-container {
  width: 70%;
  flex: 1;
  padding: 0 3rem;
  background-color: #f1f1f1;
  align-self: center;
}
