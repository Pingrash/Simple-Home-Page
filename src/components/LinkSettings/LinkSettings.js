import React from 'react';
import './LinkSettings.css';
import PropTypes from 'prop-types';

import AddLinkForm from '../AddLinkForm/AddLinkForm';
import LinkSettingCard from '../LinkSettingCard/LinkSettingCard';

const LinkSettings = ({
  links,
  toggleLinkSettings,
  addLink,
  deleteLink
}) => {
  const linkCardsToRender = links.map(link => {
    return (
      <LinkSettingCard
        key={link.name}
        link={link}
        deleteLink={deleteLink}
        data-test='setting-card'
      />
    );
  });

  return (
    <div className='settings-container'>
      <button
        onClick={() => toggleLinkSettings(false)}
        className='close-button'
        data-test='close-button'
      >
        Close
      </button>
      <AddLinkForm addLink={addLink} data-test='entry-form' />
      <div className='cards-container'>{linkCardsToRender}</div>
    </div>
  );
};

LinkSettings.propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      link: PropTypes.string.isRequired,
      favicon: PropTypes.string,
      name: PropTypes.string.isRequired
    })
  ).isRequired,
  toggleLinkSettings: PropTypes.func.isRequired,
  addLink: PropTypes.func.isRequired
};

export default LinkSettings;
