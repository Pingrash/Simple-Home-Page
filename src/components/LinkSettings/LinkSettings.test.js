import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { findByDataTest, checkProps } from '../../../Utilities';
import LinkSettings from './LinkSettings';

const testFunc = () => null;

const props = {
  toggleLinkSettings: testFunc,
  addLink: testFunc,
  links: [{ link: 'link', favicon: 'favicon', name: 'name' }],
  deleteLink: testFunc
};

const setup = () => {
  const component = shallow(<LinkSettings {...props} />);
  return component;
};

describe('Snapshot', () => {
  it('Should match snapshot', () => {
    const tree = renderer
      .create(<LinkSettings {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('LinkSettings component renders correctly', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setup();
  });

  it('Should render without errors', () => {
    expect(wrapper.length).toBe(1);
  });

  it('Should render link entry form', () => {
    const field = findByDataTest(wrapper, 'entry-form');
    expect(field.length).toBe(1);
  });

  it('Should render link setting cards', () => {
    const cards = findByDataTest(wrapper, 'setting-card');
    expect(cards.length).toBe(1);
  });

  it('Should render a close button', () => {
    const button = findByDataTest(wrapper, 'close-button');
    expect(button.length).toBe(1);
  });
});

describe('LinkSettings should have correct props', () => {
  it('Should have correct props', () => {
    const propsError = checkProps(LinkSettings, props);
    expect(propsError).toBeUndefined();
  });
});
