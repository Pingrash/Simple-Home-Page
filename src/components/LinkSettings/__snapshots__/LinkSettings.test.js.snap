// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Should match snapshot 1`] = `
<div
  className="settings-container"
>
  <button
    className="close-button"
    data-test="close-button"
    onClick={[Function]}
  >
    Close
  </button>
  <form
    className="add-form"
    data-test="entry-form"
    onSubmit={[Function]}
  >
    <h3>
      Add Link
    </h3>
    <label
      for="url"
    >
      URL: 
    </label>
    <input
      data-test="form-field"
      defaultValue="https://"
      id="url"
      onChange={[Function]}
      placeholder="https://"
      required={true}
      title="Website URL eg. https://www.google.com"
      type="url"
    />
    <label
      for="name"
    >
      Link Name: 
    </label>
    <input
      data-test="form-field"
      id="name"
      onChange={[Function]}
      required={true}
    />
    <button
      className="submit-button"
      data-test="submit-button"
      type="submit"
    >
      Submit
    </button>
  </form>
  <div
    className="cards-container"
  >
    <div
      className="card-container"
      data-test="container"
    >
      <img
        alt="favicon"
        className="card-favicon"
        data-test="favicon"
        src="favicon"
      />
      <h4
        className="card-title"
        data-test="title"
      >
        name
      </h4>
      <span
        className="card-url"
        data-test="url"
      >
        link
      </span>
      <button
        className="card-button"
        data-test="delete-button"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;
